pallets/messages/src/types.rs

```diff
@@ -43,33 +43,12 @@ where
 
 /// Newtype for CID addresses. We can change the inner type as we understand
 /// more about how to structure IPFS addressable content.
-#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
-pub struct CID(Vec<u8>);
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct FrequencyCid(Vec<u8>);
 
-impl CID {
-	/// Constructor
-	pub fn new(vec: Vec<u8>) -> CID {
-		CID(vec)
-	}
-
-	/// Exposes the inner vector
-	pub fn get(&self) -> &Vec<u8> {
-		&self.0
-	}
-}
-
-/// IPFS payloads are defined by two members, a CID address and a payload length
-#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
-pub struct IPFSPayload {
-	/// IPFS Content address
-	pub cid: CID,
-	/// Size of the IPFS payload
-	pub payload_length: u32,
-}
-
-impl IPFSPayload {
-	/// Constructor
-	pub fn new(cid: CID, payload_length: u32) -> IPFSPayload {
-		IPFSPayload { cid, payload_length }
+impl FrequencyCid {
+	/// Checks if a given string is a valid CID
+	pub fn is_valid(&self) -> bool {
+		unimplemented!("No way to check CID validity yet")
```

Comment:
The `rust-cid` crate requires `std`, which we can't work with in the pallet code. Working with the crate has been pretty time intensive -- if we need CID validation it might be worth creating a dedicated story for it.

Reply:
I like this Idea. You can remove this code and add a ticket for it.